apps/backend/src/modules/gradtrak/typedefs/gradtrak.ts

```diff
@@ -0,0 +1,235 @@
+    HAAS_PV
+
+    "HAAS Physical Science"
+    HAAS_PS
+    
+    "HAAS Social and Behavioral Sciences"
+    HAAS_SBS
+  }
+
+  type Major_Req {
+    name: String!
+    major: String!
+    num_courses_required: Int!
+    satisfying_course_ids: [String!]
+    is_minor: Boolean!
+  }
+
+  type Gradtrak {
+    user_email: String!
+    planTerms: [PlanTerm!]
+    miscellaneous: PlanTerm!
+    uni_reqs: [String!]
+    college_reqs: [String!]
+    major_reqs: [Major_Req!]
+    created: String!
+    revised: String!
+  }
+
+  type PlanTerm {
+    _id: ID
+    name: String
+    user_email: String!
+    term: GqlTerm
+    courses: [SelectedCourse!]
+    custom_events: [CustomEvent!]
+  }
+
+  type SelectedCourse {
+    """
+    Identifiers (probably cs-course-ids) for the classes the user has added to their schedule.
+    """
+    class_ID: String!
+    """
+    Identifiers (probably the "003" in "2022 Spring STAT 97 003") for the primary sections (typically lectures) the user has added to their schedule.
+    """
+    primary_section_ID: String
+    """
+    Identifiers (probably the "103" in "103 DIS") for the secondary sections (typically discussions) the user has added to their schedule.
+    """
+    secondary_section_IDs: [String!]
+    """
+    Requirements satisfied by class
+    """
+    uni_reqs: [String!]
+    college_reqs: [String!]
+  }
+
+  type CustomEvent {
+    start_time: String!
+    end_time: String!
+    title: String
+    location: String
+    description: String
+    days_of_week: String
+    uni_reqs: [String!]
+    college_reqs: [String!]
+  }
+
+  input MajorReqInput {
+    name: String!
+    major: String!
+    num_courses_required: Int!
+    satisfying_course_ids: [String!]!
+    is_minor: Boolean! 
+  }
+
+  input GqlTermInput {
+    year: Int!
+    planTerm: String!
+  }
+
+  input CustomEventInput {
+    start_time: String!
+    end_time: String!
+    title: String
+    location: String
+    description: String
+    days_of_week: String
+    uni_reqs: [Uni_Reqs!]
+    college_reqs: [College_Reqs!]
+  }
+
+  input SelectedCourseInput {
+    class_ID: String!
+    primary_section_ID: String
+    secondary_section_IDs: [String!]
+    uni_reqs: [Uni_Reqs!]
+    college_reqs: [College_Reqs!]
+  }
+
+  input PlanTermInput {
+    name: String
+    term: GqlTermInput
+    courses: [SelectedCourseInput!]
+    custom_events: [CustomEventInput!]
+  }
+
+  type Query {
+    """
+    Takes in user's email and returns their entire gradtrak
+    """
+    gradtrakByUser: Gradtrak @auth
+    
+    """
+    Takes in an ID and returns the planTerm for that term.
+    """
+    planTermByID(id: ID!): PlanTerm
+  }
+
+  type Mutation {
+    """
+    Takes in user's email, creates a new Gradtrak record in the database, and returns the Gradtrak
+    """
+    createNewGradtrak: Gradtrak @auth
+
+    """
+    Takes in a new college, switches the user's college requirements
+    """
+    changeGradtrakCollege(college: Colleges!): Gradtrak @auth
```

Comment:
I would suggest switching to a generic `updatePlan` that takes in a new college and any other plan-specific fields rather than an endpoint for every action.